File: src/main/java/com/eager/questioncloud/creator/CreatorJpaRepository.java
package com.eager.questioncloud.creator;

import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface CreatorJpaRepository extends JpaRepository<CreatorEntity, Long> {
    Boolean existsByUserId(Long userId);

    Optional<CreatorEntity> findByUserId(Long userId);
}

File: qc-domain/qc-creator/src/main/kotlin/com/eager/questioncloud/creator/infrastructure/repository/CreatorJpaRepository.kt
package com.eager.questioncloud.creator.infrastructure.repository

import com.eager.questioncloud.creator.infrastructure.entity.CreatorEntity
import org.springframework.data.jpa.repository.JpaRepository

interface CreatorJpaRepository : JpaRepository<CreatorEntity, Long> {
    fun findByIdIn(creatorIds: List<Long>): List<CreatorEntity>

    fun findByUserId(userId: Long): CreatorEntity?
}
